### PR TITLE
Restore TS types

### LINKED ISSUE #1036: TS types
hey @kibertoad PR #913 was reverted for reasons listed in the revert PR #1035 

would you mind taking another run at it with a new PR

### PR DESCRIPTION
**Pull request by kibertoad**
fixes #1036

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -19,8 +22,9 @@
     "docs": "docsify serve",
     "browser-test": "airtap --local 8080 test/browser*test.js",
     "lint": "eslint .",
-    "test": "npm run lint && tap --100 test/*test.js test/*/*test.js",
-    "test-ci": "npm run lint && tap test/*test.js test/*/*test.js --coverage-report=lcovonly",
+    "test": "npm run lint && tap --100 test/*test.js test/*/*test.js && npm run test-types",
+    "test-ci": "npm run lint && tap test/*test.js test/*/*test.js --coverage-report=lcovonly && npm run test-types",
```
Comment (kibertoad) [inline package.json:26]: @davidmarkclements Please let me know if you want this removed.

Comment (davidmarkclements) [reply package.json:26]: yeah can you please remove test-ci and make it work as before with your additions

Comment (kibertoad) [reply package.json:26]: @davidmarkclements Do you mean removing entire "test-ci", or just "npm run test-types" out of it?

**Comment by davidmarkclements** [comment]
when I run `npm run test-types` on the `next` branch prior to the revert I get this:

image


This doesn't happen on your branch but can you also please make sure this isn't an issue

**Review by davidmarkclements — CHANGES_REQUESTED** [review]
as commented

**Comment by kibertoad** [comment]
@davidmarkclements Tests are fixed already in this branch. Could you please clarify what specifically do you want me to do with `test-ci`? It existed prior to "TS type" PR, and in my original PR I didn't touch it at all, which is why I'm confused why it was mentioned in the first place.

**Comment by davidmarkclements** [comment]
@kibertoad Oh I see - my apologies I thought it was added in the PR 

Ok I'll merge this and check the `next` branch now

**Comment by davidmarkclements** [comment]
perfect - all good. Thanks @kibertoad.

@mcollina @jsumners - I think "test-ci" is a bad idea - it allows for issues in the `npm test` flow

**Comment by mcollina** [comment]
I agree, we should remove it

**Pull request merged**